src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/integration/JwtAuthHelper.kt

```diff
@@ -0,0 +1,65 @@
+package uk.gov.justice.digital.hmpps.digitalprisonreportingmi.integration
+
+import io.jsonwebtoken.Jwts
+import io.jsonwebtoken.SignatureAlgorithm
+import org.springframework.context.annotation.Bean
+import org.springframework.http.HttpHeaders
+import org.springframework.security.oauth2.jwt.JwtDecoder
+import org.springframework.security.oauth2.jwt.NimbusJwtDecoder
+import org.springframework.stereotype.Component
+import java.security.KeyPair
+import java.security.KeyPairGenerator
+import java.security.interfaces.RSAPublicKey
+import java.time.Duration
+import java.util.Date
+import java.util.UUID
+
+@Component
+class JwtAuthHelper {
+  private val keyPair: KeyPair
+
+  init {
+    val gen = KeyPairGenerator.getInstance("RSA")
+    gen.initialize(2048)
+    keyPair = gen.generateKeyPair()
+  }
+
+  @Bean
+  fun jwtDecoder(): JwtDecoder = NimbusJwtDecoder.withPublicKey(keyPair.public as RSAPublicKey).build()
+
+  fun setAuthorisation(
+    user: String = "prison-reporting-mi-client",
+    roles: List<String> = listOf(),
+    scopes: List<String> = listOf(),
+  ): (HttpHeaders) -> Unit {
+    val token = createJwt(
+      subject = user,
+      scope = scopes,
+      expiryTime = Duration.ofHours(1L),
+      roles = roles,
+    )
+    return { it.set(HttpHeaders.AUTHORIZATION, "Bearer $token") }
+  }
+
+  internal fun createJwt(
+    subject: String?,
+    scope: List<String>? = listOf(),
+    roles: List<String>? = listOf(),
+    expiryTime: Duration = Duration.ofHours(1),
+    jwtId: String = UUID.randomUUID().toString(),
+  ): String =
+    mutableMapOf<String, Any>()
```

Comment:
If we can be sure we'll always have a value for roles and scope we can then tidy the following code up so we can
* use an immutable map e.g. mapOf(.....)
* remove the nested use of let inside also (imho this is a bit hard to read)

Reply:
This has now been changed